apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -101,11 +215,23 @@ def replace(
 
             >>> context.replace(tiledb_config={"vfs.s3.region": "us-east-2"})
         """
-        if tiledb_config:
-            new_config = self.tiledb_ctx.config()
-            new_config.update(tiledb_config)
-            changes["tiledb_ctx"] = tiledb.Ctx(config=new_config)
-        return attrs.evolve(self, **changes)
+        with self._lock:
+            if tiledb_config is not None:
+                if tiledb_ctx:
+                    raise ValueError(
+                        "Either tiledb_config or tiledb_ctx may be provided"
+                        " to replace(), but not both."
+                    )
+                new_config = dict()
```

Comment:
typo?  I think you want to delete this line in favor of the next line

Reply:
Whoops! This is a remnant of a previous version (where `_internal_tiledb_config` was not guaranteed to return a fresh dict).